roles/openshift_health_checker/openshift_checks/__init__.py

```diff
@@ -62,6 +69,45 @@ def subclasses(cls):
             for subclass in subclass.subclasses():
                 yield subclass
 
+    def execute_module(self, *args, **kwargs):
+        """Invoke an Ansible module from a check.
+
+        This is the same as invoking _execute_module from an action plugin
+        but task_vars and tmp from the check are used unless specified.
+        So e.g. check.execute_module(module_name="foo", module_args=dict(arg1=...))
+        Return: result hash from module execution.
+        """
+
+        # fill in unspecified task_vars as either positional (3rd) or keyword arg
+        if len(args) > 2 and args[2] is None:
+            args[2] = self.task_vars
+        else:
+            kwargs["task_vars"] = kwargs.get("task_vars", self.task_vars)
+
+        # fill in unspecified tmp as either positional (4th) or keyword arg
+        if len(args) > 3 and args[3] is None:
+            args[3] = self.tmp
+        else:
+            kwargs["tmp"] = kwargs.get("tmp", self.tmp)
```

Comment:
Couldn't we simply replicate the signature of the Ansible method here and then override `task_vars` and `tmp`?

```python
def execute_module(self, module_name=None, module_args=None, tmp=None, task_vars=None, persist_files=False, delete_remote_tmp=True, wrap_async=False):
    if tmp is not None:
        raise RuntimeError('Usage error: tmp is ignored by this method, use _execute_module instead.')
    if task_vars is not None:
        raise RuntimeError('Usage error: task_vars is ignored by this method, use _execute_module instead.')
    return self._execute_module(module_name, module_args, self.tmp, self.task_vars, persist_files, delete_remote_tmp, wrap_async)